pkgs/development/libraries/xsimd/default.nix

```diff
@@ -1,35 +1,32 @@
-{ lib, stdenv, fetchFromGitHub, cmake, gtest }:
+{ lib, stdenv, fetchFromGitHub, cmake, doctest }:
 stdenv.mkDerivation rec {
   pname = "xsimd";
-  version = "9.0.1";
+  version = "10.0.0";
   src = fetchFromGitHub {
     owner = "xtensor-stack";
     repo = "xsimd";
     rev = version;
-    sha256 = "sha256-onALN6agtrHWigtFlCeefD9CiRZI4Y690XTzy2UDnrk=";
+    sha256 = "sha256-+ewKbce+rjNWQ0nQzm6O4xSwgzizSPpDPidkQYuoSTU=";
   };
 
   nativeBuildInputs = [ cmake ];
 
   cmakeFlags = [ "-DBUILD_TESTS=ON" ];
 
   doCheck = true;
-  nativeCheckInputs = [ gtest ];
-  checkTarget = "xtest";
-  GTEST_FILTER =
+  nativeCheckInputs = [ doctest ];
+  checkPhase =
     let
       # Upstream Issue: https://github.com/xtensor-stack/xsimd/issues/456
-      filteredTests = lib.optionals stdenv.hostPlatform.isDarwin [
-        "error_gamma_test/*"
+      filteredTests = [
+        "error_gamma_test"
       ];
     in
-    "-${builtins.concatStringsSep ":" filteredTests}";
-
-  # https://github.com/xtensor-stack/xsimd/issues/748
-  postPatch = ''
-    substituteInPlace xsimd.pc.in \
-      --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@
-  '';
+    ''
+      runHook preCheck
+      ./test/test_xsimd --source-file-exclude=${builtins.concatStringsSep "," filteredTests}
+      runHook postCheck
+    '';
```

Comment:
Wait, this `--source-file-exclude` appears to be a no-op, I'm not sure what names I need to pass it